resources/sdk/pureclouddotnet/extensions/Extensions/AuthExtensions.cs

```diff
@@ -85,9 +97,18 @@ public static AuthTokenInfo PostToken(this ApiClient apiClient, string clientId,
                 throw new ApiException(statusCode, "Error calling PostToken: " + response.ErrorMessage,
                     response.ErrorMessage);
 
+            var authTokenInfo = (AuthTokenInfo) apiClient.Deserialize(response, typeof (AuthTokenInfo));
+
+            if (apiClient.UsingCodeAuth) {
```

Comment:
Let's remove all the code in this if block. The .NET SDK does not mutate the ApiClient instance when performing authorization (others do, so this feedback only applies here). The consuming application is expected to keep track of the refresh token and credentials. There's a few lines above (line 40) that should be removed as well. Instead, use a local variable to track if it's a code auth request.

Reply:
I just realized why you did this; the implementation in ApiClient to automatically get a refresh token requires it. Because of that, let's change this to store the auth response in all cases, not just for code auth grants. I'd suggest removing the individual properties and just adding an `AuthTokenInfo` property.